modules/ztp-nbde-generating-a-new-tang-server-key.adoc

```diff
@@ -0,0 +1,120 @@
+// Module included in the following assemblies:
+//
+// scalability_and_performance/ztp-nbde-implementation-guide.adoc
+
+[id="ztp-nbde-generating-a-new-tang-server-key_{context}"]
+= Generating a new Tang server key
+
+.Prerequisites
+
+* A root shell on the Linux machine running the Tang server.
+
+* To facilitate verification of the Tang server key rotation, encrypt a small test file with the old key:
++
+[source,terminal]
+----
+# echo plaintext | clevis encrypt tang '{"url":"http://localhost:7500”}' -y >/tmp/encrypted.oldkey
+----
++
+[source,terminal]
+----
+# clevis decrypt </tmp/encrypted.oldkey plaintext
+----
+
+.Procedure
+
+. Locate and access the directory that stores the Tang server key. This is usually the `/var/db/tang` directory:
++
+[source,terminal]
+----
+# tang-show-keys 7500
+----
++
+.Example output
+
+[source,terminal]
+----
+36AHjNH3NZDSnlONLz1-V4ie6t8
+----
++
+[source,terminal]
+----
+# cd /var/db/tang/
+----
++
+[source,terminal]
+----
+# ls -A1
+----
++
+.Example output
+[source,terminal]
+----
+36AHjNH3NZDSnlONLz1-V4ie6t8.jwk
+gJZiNPMLRBnyo_ZKfK4_5SrnHYo.jwk
+----
++
+During normal Tang server operations, there are two `.jwk` files in this directory: one for signing and verification, and another for key derivation.
+
+. Disable advertisement of the old keys:
++
+[source,terminal]
+----
+# for key in *.jwk; do \
+  mv -- "$key" ".$key"; \
+done
+----
++
+New clients setting up Network-Bound Disk Encryption (NBDE) or requesting keys will no longer see the old keys. Existing clients can still access and use the old keys until they are deleted. The Tang server reads but does not advertise keys stored in UNIX hidden files, which start with the `.` character.
+
+. Generate a new key:
++
+[source,terminal]
+----
+# /usr/libexec/tangd-keygen /var/db/tang
+----
++
+[source,terminal]
+----
+# ls -A1
+----
++
+.Example output
+[source,terminal]
+----
+.36AHjNH3NZDSnlONLz1-V4ie6t8.jwk
+.gJZiNPMLRBnyo_ZKfK4_5SrnHYo.jwk
+Bp8XjITceWSN_7XFfW7WfJDTomE.jwk
+WOjQYkyK7DxY_T5pMncMO5w0f6E.jwk
+----
++
+Tang automatically advertises the new keys.
++
+[NOTE]
+====
+More recent Tang server installations include a helper `/usr/libexec/tangd-rotate-keys` directory that takes care of disabling advertisement and generating the new keys simultaneously.
+====
+
+. If you are running multiple Tang servers behind a load balancer that share the same key material, ensure the changes made here are properly synchronized across the entire set of servers before proceeding.
+
+.Verification
+
+. Verify that the Tang server is advertising the new key, and not advertising the old key:
++
+[source,terminal]
+----
+# tang-show-keys 7500
+----
++
+[source,terminal]
+----
+WOjQYkyK7DxY_T5pMncMO5w0f6E
+----
+
+. Verify that the old key, while not advertised, is still available to decryption requests:
++
+[source,terminal]
+----
+# clevis decrypt </tmp/encrypted.oldkey
+plaintext
```

Comment:
`plaintext` is the output, not part of the command

Reply:
removed